src/libraries/Common/tests/System/Net/Http/TestHelper.cs

```diff
@@ -130,6 +130,23 @@ public static void EnableUnencryptedHttp2IfNecessary(HttpClientHandler handler)
             object socketsHttpHandler = socketsHttpHandlerField.GetValue(handler);
             Assert.NotNull(socketsHttpHandler);
 
+            EnableUncryptedHttp2(socketsHttpHandler);
+        }
+
+#if !NETFRAMEWORK
```

Comment:
Why do we need this? Should we fix either PlatformDetection or the capability Instead?

Reply:
`SocketsHttpHandler` type is not available on the full .Net Framework, so conditional compilation seems to be the only way.

Reply:
So why type it as `SocketsHttpHandler` rather than `object`, as the other functions this is called from and it's calling is doing?

Reply:
It's to distinct it from the [other overload taking HttpClientHandler](https://github.com/dotnet/runtime/blob/cb60aaa2408f8423d95d3ef03ae5b6f372a1945b/src/libraries/Common/tests/System/Net/Http/TestHelper.cs#L116).

Reply:
Of course, I could have made it `object`, but think an explicit type is more convenient.